modules/fluid_properties/src/userobjects/SalineFluidProperties.C

```diff
@@ -0,0 +1,146 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+//
+#include "SalineFluidProperties.h"
+
+registerMooseObject("FluidPropertiesApp", SalineFluidProperties);
+
+InputParameters
+SalineFluidProperties::validParams()
+{
+  InputParameters params = SinglePhaseFluidProperties::validParams();
+  params.addRequiredParam<std::string>(
+      "comp_name", "The name of the components in the salt (e.g., 'LiF-NaF-KF')");
+  params.addRequiredParam<std::string>(
+      "comp_val", "The mole fraction of each salt component (e.g., '0.5-0.1-0.4)");
+  params.addRequiredParam<std::string>(
+      "prop_def",
+      "Definition of a fluid property file, which must be a file path to the "
+      "comma-separated data matching the Saline format.");
+  return params;
+}
+
+SalineFluidProperties::SalineFluidProperties(const InputParameters & parameters)
+  : SinglePhaseFluidProperties(parameters)
+{
+
+#ifdef SALINE_ENABLED
+
+  std::string propDef = getParam<std::string>("prop_def");
+  _d.load(propDef);
+  bool success = _tp.initialize(&_d);
+  if (!success)
+    mooseError("The initialize of the Saline interface has failed");
+  std::string name = getParam<std::string>("comp_name");
+  std::string comp = getParam<std::string>("comp_val");
+  std::vector<std::string> nameList;
+  MooseUtils::tokenize<std::string>(name, nameList, 1, "-");
+  std::vector<Real> valList;
+  MooseUtils::tokenizeAndConvert<Real>(comp, valList, "-");
+  Real mole_sum = 0.0;
+  for (Real val : valList)
+    mole_sum = mole_sum + val;
+  if (std::abs(mole_sum - 1.0) > 1e-6)
```

Comment:
I'm not a fan of a hardcoded custom tolerance here. You could use `MooseUtils::absoluteFuzzyEqual(mole_sum, 1.0)`

Reply:
This will be included in the next set of changes.

Reply:
@rks171

Reply:
fixed